vscode/src/commands/cache.ts

```diff
@@ -0,0 +1,61 @@
+/*
+  Copyright (c) 2023-2024, Oracle and/or its affiliates.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import { commands, window } from "vscode";
+import { globalVars } from "../extension";
+import { builtInCommands, extCommands } from "./commands";
+import { ICommand } from "./types";
+import { l10n } from "../localiser";
+import * as fs from 'fs';
+import * as path from 'path';
+
+const deleteCache = async () => {
+    // TODO: Change workspace path to userdir path
+    const storagePath = globalVars.extensionInfo.getWorkspaceStorage()?.fsPath;
+    if (!storagePath) {
+        window.showErrorMessage(l10n.value("jdk.extension.cache.error_msg.cannotFindWrkSpacePath"));
+        return;
+    }
+
+    const userDir = path.join(storagePath, "userdir");
+    if (userDir && fs.existsSync(userDir)) {
+        const yes = l10n.value("jdk.extension.cache.label.confirmation.yes")
+        const cancel = l10n.value("jdk.extension.cache.label.confirmation.cancel")
+        const confirmation = await window.showInformationMessage('Are you sure you want to delete cache for this workspace  and reload the window ?',
+            yes, cancel);
```

Comment:
I think the confirmation message needs to be replaced by its localised key lookup`l10n.value("jdk.extension.cache.message.confirmToDeleteCache")` and the message value may be updated for this key in *bundle.l10n.en.json*.

Reply:
Done